app/helpers/interview_questions_helper.rb

```diff
@@ -0,0 +1,13 @@
+module InterviewQuestionsHelper
+  def color_link(question)
+    color = "bg_link"
+    color += " bg_" + question.mark unless question.mark.blank?
+    color
+  end
+
+  def button_next(interview, question)
+    next_but = "NEXT"
+    next_but = "FINISH" if (interview.interview_questions.count - 1 + interview.interview_questions.first.id) == question.id
```

Comment:
1. You can not deduct record id from records count. It is same as to deduct kilograms from kilometres.
2. Please explain what is view layer logic and business layer logic and where should be situated code related to business and view layer?
3. According to your explanation find out what's wrong with your code.